src/core/server/saved_objects/migrationsv2/model/helpers.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { gt, valid } from 'semver';
+import { State } from '../types';
+import { IndexMapping } from '../../mappings';
+import { FetchIndexResponse } from '../actions';
+
+/**
+ * A helper function/type for ensuring that all control state's are handled.
+ */
+export function throwBadControlState(p: never): never;
+export function throwBadControlState(controlState: any) {
+  throw new Error('Unexpected control state: ' + controlState);
```

Comment:
All the file was extracted from `model.ts`

Reply:
Same [comment](https://github.com/elastic/kibana/pull/103341/files?w=1#r660163075) about doing too much in one PR.